Implementation for Service accounts

prashanthca opened the pull request:
This code change will allow the usage of service accounts with GDIndex. This will mainly benefit the end-users

maple3142 commented:
Looks good, but I don't really what this is for. Also, add a brief documentation to `README.md` might be necessary, or users will not know this feature.

As for code addition, please fix some coding style using prettier with [this](https://github.com/maple3142/dotfiles/blob/master/.prettierrc) config. Also, prefer `const` over `let`, `var` if not necessary.

CatTheHacker commented:
Service Accounts can be used instead of standard Google/GSuite accounts with `rclone` setup.
The same way `rclone` can use Service Accounts - https://rclone.org/drive/#service-account-support.

prashanthca commented:
Thanks for looking into this @maple3142! And yes, as @CatTheHacker said, it's basically a way to use service accounts to access GDrive. This eliminates the standard upload/download quota that usually applies to normal accounts

CatTheHacker commented:
> This eliminates the standard upload/download quota that usually applies to normal accounts

Service accounts are still subject to upload/download quota just like the normal user accounts, it's just easier and faster to manage/use as they don't require manual authentication via web browser.

prashanthca commented:
@CatTheHacker Yes, it seems that I was mistaken about the quota. Nevertheless, as you mentioned, it's a easier way to setup GDIndex, without having to get the refresh token via rclone or the web method. Maybe I can take a look into modifying the props to contain multiple service account credentials, so the client can cycle through them randomly, distributing the usage across the accounts.

prashanthca commented:
@maple3142, I've made the changes you suggested and I've added a section in README for this feature. I've made changes in the Chinese README (both traditional and simplified versions) files as well.

maple3142 commented on worker/package.json:
```diff
@@ -5,10 +5,14 @@
 	"devDependencies": {
-		"bili": "^4.8.1"
+		"bili": "^4.8.1",
+		"rollup-plugin-node-builtins":"*",
```
Please don't use "*" as version number.
Just use what appears after running `npm install -S $pkg` or `yarn install $pkg`

maple3142 replied:
Others packages using "*" should be fixed as well. As long as this is fixed, I can merge that PR now.

prashanthca replied:
I've pushed the fix
